test/unit/org/apache/cassandra/guardrails/GuardrailTester.java

```diff
@@ -86,6 +84,19 @@ public void beforeGuardrailTest() throws Throwable
         useSuperUser();
         executeNet(format("CREATE USER IF NOT EXISTS %s WITH PASSWORD '%s'", USERNAME, PASSWORD));
         executeNet(format("GRANT ALL ON KEYSPACE %s TO %s", KEYSPACE, USERNAME));
+
+        boolean permissionApplied = false;
```

Comment:
+1 with a small change - try to use `Awaitility` which seems to be designed for this kind of stuff, an maybe use 10 seconds timeout

Reply:
TIL - Awaitility - Thanks for the pointer! 
I have made an update to use this.